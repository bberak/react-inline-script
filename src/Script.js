import React, { Component } from "react";

export default class Script extends Component {
  render() {
    let src = this.props.children || this.props.src || this.props.source || "";
    let props = {...this.props, children: null, dangerouslySetInnerHTML: {__html: src} };
    return <script {...props} /> 
  }
}