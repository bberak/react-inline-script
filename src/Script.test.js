import React from 'react';
import Script from './Script';
import renderer from 'react-test-renderer';

it('renders correctly without any source code', () => {
  const tree = renderer
    .create(<Script />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with unquoted text as a child', () => {
  const tree = renderer
    .create(<Script>alert("Howdy")</Script>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});