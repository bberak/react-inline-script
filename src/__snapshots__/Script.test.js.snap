// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with unquoted text as a child 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "alert(\\"Howdy\\")",
    }
  }
/>
`;

exports[`renders correctly without any source code 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "",
    }
  }
/>
`;
